x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java

```diff
@@ -322,6 +326,16 @@ public Settings additionalSettings() {
         return handlers;
     }
 
+    @Override
+    public List<ExecutorBuilder<?>> getExecutorBuilders(final Settings settings) {
+        final int allocatedProcessors = EsExecutors.allocatedProcessors(settings);
+        return List.of(
+                new FixedExecutorBuilder(settings, SecurityField.SECURITY_CRYPTO_THREAD_POOL_NAME,
+                        (allocatedProcessors + 1) / 2, 1000,
+                        "xpack.security.crypto.thread_pool", false)
```

Comment:
We have this threadpool for "crypto" that we use for pbkdf2 hashes in Security for API Keys (at least), which I'm now pulling into xpack core to attempt to reuse it for some pbkdf2 hashes in the encrypted repo plugin.

But, in this case, the `xpack.crypto` is not really a suitable prefix. It's unfortunate, but it hasn't bothered me enough to look into a way to rename thread pools. IMO if you're concerned with settings like this, you're diving into internals enough to not be bothered by incongruous names.

Feel free to question that.
I even wouldn't object much to just using the generic thread pool instead.